p2p/protocol/identify/id.go

```diff
@@ -395,28 +399,35 @@ func (ids *IDService) sendIdentifyResp(s network.Stream) {
 	}
 
 	ph.snapshotMu.RLock()
-	mes := &pb.Identify{}
-	ids.populateMessage(mes, c, ph.snapshot)
-	w := ggio.NewDelimitedWriter(s)
-	w.WriteMsg(mes)
-
+	ids.writeChunkedIdentifyMsg(c, ph.snapshot, s)
 	log.Debugf("%s sent message to %s %s", ID, c.RemotePeer(), c.RemoteMultiaddr())
 }
 
 func (ids *IDService) handleIdentifyResponse(s network.Stream) {
 	c := s.Conn()
 
-	r := ggio.NewDelimitedReader(s, 2048)
+	r := ggio.NewDelimitedReader(s, signedIDSize)
 	mes := pb.Identify{}
 	if err := r.ReadMsg(&mes); err != nil {
 		log.Warning("error reading identify message: ", err)
 		s.Reset()
 		return
 	}
 
+	if mes.More {
+		m := &pb.Identify{}
```

Comment:
I don't think we need to make a new message here, just reuse the existing one and merge in the new fields.

Reply:
This will also generalize this code to merge protocols, addresses, etc. We can write this as:

```go
func mergeMessages(r ggio.Reader, m proto.Message) error {
	for i := 0; i < maxMessages; i++ {
	  	switch err := r.ReadMsg(mes); err {
	    default:
		case io.EOF:
        	return nil
	    case nil:
			log.Warning("error reading identify message: ", err)
			s.Reset()
			return
		}
	}
    return fmt.Errorf("too many parts")
}
```

Reply:
@Stebalien 

Had to use `proto.Merge(dst, src)` here after calling the goog reader because  the gogo varint reader uses `proto.Unmarshal` under the hood which resets(clears) the existing proto message before reading the new one into it.

Other than that, works well.